pypet2bids/pypet2bids/ecat2nii.py

```diff
@@ -248,66 +241,15 @@ def ecat2nii(ecat_main_header=None,
             step_name='10_save_nii_ecat2nii_python'
         )
 
-    # populating nifti header
-    if img_nii.header['sizeof_hdr'] != 348:
-        img_nii.header['sizeof_hdr'] = 348
-    # img_nii.header['dim_info'] is populated on object creation
-    # img_nii.header['dim']  is populated on object creation
-    img_nii.header['intent_p1'] = 0
-    img_nii.header['intent_p2'] = 0
-    img_nii.header['intent_p3'] = 0
-    # img_nii.header['datatype'] # created on invocation seems to be 16 or int16
-    # img_nii.header['bitpix'] # also automatically created and inferred 32 as of testing w/ cimbi dataset
-    # img_nii.header['slice_type'] # defaults to 0
-    # img_nii.header['pixdim'] # appears as 1d array of length 8 we rescale this
-    img_nii.header['pixdim'] = numpy.array(
-        [1,
-         sub_headers[0]['X_PIXEL_SIZE'] * 10,
-         sub_headers[0]['Y_PIXEL_SIZE'] * 10,
-         sub_headers[0]['Z_PIXEL_SIZE'] * 10,
-         0,
-         0,
-         0,
-         0])
-    img_nii.header['vox_offset'] = 352
-
-    # TODO img_nii.header['scl_slope'] # this is a NaN array by default but apparently it should be the dose calibration
-    #  factor img_nii.header['scl_inter'] # defaults to NaN array
-    img_nii.header['scl_slope'] = 1
-    img_nii.header['scl_inter'] = 0
-    img_nii.header['slice_end'] = 0
-    img_nii.header['slice_code'] = 0
-    img_nii.header['xyzt_units'] = 10
+    img_nii.header.set_slope_inter(slope=1, inter=0)
+    img_nii.header.set_xyzt_units('mm', 'unknown')
+    img_nii.header.set_qform(affine, code=1)
```

Comment:
If you're going to go through the trouble of setting the qform, you might as well say that it's there. 0 means invalid, not "arbitrary", as the comment previously said.

Reply:
No arguments from me.